google/auth/transport/aiohttp_req.py

```diff
@@ -0,0 +1,414 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Transport adapter for Async HTTP (aiohttp)."""
+
+from __future__ import absolute_import
+
+import asyncio
+import functools
+import logging
+import numbers
+import time
+
+import aiohttp
+import six
+
+# import google.auth
+from google.auth import exceptions
+from google.auth import transport
+
+# import google.auth.transport._mtls_helper
+
+
+_OAUTH_SCOPES = [
+    "https://www.googleapis.com/auth/appengine.apis",
+    "https://www.googleapis.com/auth/userinfo.email",
+]
+
+_LOGGER = logging.getLogger(__name__)
+
+_DEFAULT_TIMEOUT = 120  # in seconds
+
+
+class _Response(transport.Response):
+    """
+    Requests transport response adapter.
+
+    Args:
+        response (requests.Response): The raw Requests response.
+    """
+
+    def __init__(self, response):
+        self.response = response
+
+    @property
+    def status(self):
+        return self.response.status
+
+    @property
+    def headers(self):
+        return self.response.headers
+
+    @property
+    def data(self):
+        return self.response.content
+
+
+class TimeoutGuard(object):
+    """A context manager raising an error if the suite execution took too long.
+
+    Args:
+        timeout ([Union[None, float, Tuple[float, float]]]):
+            The maximum number of seconds a suite can run without the context
+            manager raising a timeout exception on exit. If passed as a tuple,
+            the smaller of the values is taken as a timeout. If ``None``, a
+            timeout error is never raised.
+        timeout_error_type (Optional[Exception]):
+            The type of the error to raise on timeout. Defaults to
+            :class:`requests.exceptions.Timeout`.
+    """
+
+    def __init__(self, timeout, timeout_error_type=asyncio.TimeoutError):
+        self._timeout = timeout
+        self.remaining_timeout = timeout
+        self._timeout_error_type = timeout_error_type
+
+    def __enter__(self):
+        self._start = time.time()
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        if exc_value:
```

Comment:
this seems to take from transport/request.py. can any of the code be shared as to not copy/paste?

Also, filename is `aiohttp_req.py`. Generally it is best to not abbreviate, but it can be acceptable. Is there a reason to abbreviate request?

Reply:
Resolved